Weather_App/Weather_App/Data/Gateways/WeatherGateway.swift

```diff
@@ -11,9 +11,32 @@ import RxSwift
 protocol WeatherGatewayType {
     func getWeatherCurrent(latitude: Double, longitude: Double) -> Observable<WeatherCurrent>
     func getWeatherForecast(latitude: Double, longitude: Double) -> Observable<WeatherForecast>
+    
+    func getFavoriteWeatherCurrent() -> Observable<[WeatherCurrentEntity]>
+    func getUserLocationWeatherCurrent() -> Observable<[WeatherCurrentEntity]>
+    func saveWeatherCurrent(weatherCurrent: WeatherCurrent) -> Observable<Void>
+    func updateIsFavoriteWeatherCurrent(for cityName: String, isFavorite: Bool) -> Observable<Void>
+    func updateUserLocationWeatherCurrent(for cityName: String, userLocation: Bool) -> Observable<Void>
+    func deleteWeatherCurrent(_ entity: WeatherCurrentEntity) -> Observable<Void>
+    func deleteCurrentUserLocationWeatherCurrent() -> Observable<Void>
+    func deleteUnusedWeatherCurrent() -> Observable<Void>
+    
+    func getFavoriteWeatherForecast() -> Observable<[WeatherForecastEntity]>
+    func getUserLocationWeatherForecast() -> Observable<[WeatherForecastEntity]>
+    func saveWeatherForecast(weatherForecast: WeatherForecast) -> Observable<Void>
+    func updateIsFavoriteWeatherForecast(for cityName: String, isFavorite: Bool) -> Observable<Void>
+    func updateUserLocationWeatherForecast(for cityName: String, userLocation: Bool) -> Observable<Void>
+    func deleteWeatherForecast(_ entity: WeatherForecastEntity) -> Observable<Void>
+    func deleteCurrentUserLocationWeatherForecast() -> Observable<Void>
+    func deleteUnusedWeatherForecast() -> Observable<Void>
 }
 
 struct WeatherGateway: WeatherGatewayType {
+    let weatherCurrentRepository = WeatherRepository<WeatherCurrentEntity>()
+    let weatherForecastRepository = WeatherRepository<WeatherForecastEntity>()
```

Comment:
```suggestion
    let weatherCurrentRepository = WeatherRepository<WeatherCurrentEntity>()
    let weatherForecastRepository = WeatherRepository<WeatherForecastEntity>()
```

Repository không phụ thuộc vào entity khi khởi tạo như thế này nhé

Reply:
chỉ một Weather Repository duy nhất